Assets/Plugins/Networking/API.cs

```diff
@@ -0,0 +1,215 @@
+    }
+    private string _PlayerSessionId;
+    public string PlayerSessionId
+    {
+        get
+        {
+            return _PlayerSessionId;
+        }
+    }
+
+    private string _CharacterModelId;
+    public string CharacterModelId
+    {
+        get
+        {
+            return _CharacterModelId;
+        }
+        set
+        {
+            _CharacterModelId = value;
+        }
+    }
+
+    private string _DnsName;
+    public string DnsName
+    {
+        get
+        {
+            return _DnsName;
+        }
+        set
+        {
+            DnsName = value;
+        }
+    }
+    private API()
+    {
+        client = new HttpClient();
+    }
+
+    public async Task Login(string username, string password)
+    {
+        BLEDebug.LogInfo("Logging In");
+        FlatJSON fJSON = new FlatJSON();
+        fJSON.Add("username", username);
+        fJSON.Add("password", password);
+        HttpContent req = fJSON.SerializeContent();
+
+        var result = await client.PostAsync("https://echelon.3pointlabs.org/auth/login", req);
+        string body = await result.Content.ReadAsStringAsync();
+        BLEDebug.LogInfo("Auth " + body);
+        fJSON.Deserialize(body);
+        fJSON.TryGetStringValue("username", out _PlayerId);
+        fJSON.TryGetStringValue("idToken", out idToken);
+        fJSON.TryGetStringValue("refreshToken", out refreshToken);
+        fJSON.TryGetStringValue("accessToken", out accessToken);
+
+        // Set Token for Authentication
+        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(idToken);
+    }
+
+    public async Task CreateMatchmakingTicket()
+    {
+        BLEDebug.LogInfo("Creating MatchMaking Ticket");
+        FlatJSON fJSON = new FlatJSON();
+
+        HttpContent req = fJSON.SerializeContent();
+        HttpResponseMessage result = await client.PostAsync("https://echelon.3pointlabs.org/matchmaking/ticket", req);
+
+        string body = await result.Content.ReadAsStringAsync();
+        BLEDebug.LogInfo("Ticket" + body);
+        fJSON.Deserialize(body);
+
+        fJSON.TryGetStringValue("TicketId", out TicketId);
+        fJSON.TryGetStringValue("Status", out _Status);
+    }
+
+    public async Task CheckMatchmakingTicketStatus()
+    {
+        BLEDebug.LogInfo("Checking MatchMaking Ticket Status");
+        FlatJSON fJSON = new FlatJSON();
+        fJSON.Add("TicketId", TicketId);
+        HttpContent req = fJSON.SerializeContent();
+        HttpResponseMessage result = await client.PostAsync("https://echelon.3pointlabs.org/matchmaking/status", req);
+
+        string body = await result.Content.ReadAsStringAsync();
+        BLEDebug.LogInfo("Ticket " + body);
+        fJSON.Deserialize(body);
+
+        fJSON.TryGetStringValue("Status", out _Status);
+
+        if (Status == "COMPLETED")
+        {
+            fJSON.TryGetStringValue("GameSessionArn", out _GameSessionArn);
+            fJSON.TryGetStringValue("IpAddress", out _IpAddress);
```

Comment:
Did we want to leave IP as a string?

Reply:
Yes, because the endpoint for the GameLift connection is taken in as a string, even if it is an IP address instead of DNS.